File: app/src/main/java/com/drdisagree/iconify/utils/helpers/LocaleHelper.java
package com.drdisagree.iconify.utils.helpers;

import static com.drdisagree.iconify.common.Preferences.APP_LANGUAGE;

import android.content.Context;
import android.content.res.Configuration;
import android.os.LocaleList;

import com.drdisagree.iconify.config.Prefs;

import java.util.Locale;

public class LocaleHelper {

    public static Context setLocale(Context context) {
        String localeCode = Prefs.getString(APP_LANGUAGE, "");

        if (!localeCode.isEmpty()) {
            Locale locale = Locale.forLanguageTag(localeCode);

            Configuration configuration = context.getResources().getConfiguration();
            configuration.setLocale(locale);
            configuration.setLayoutDirection(locale);

            LocaleList localeList = new LocaleList(locale);
            LocaleList.setDefault(localeList);
            configuration.setLocales(localeList);

            context = context.createConfigurationContext(configuration);
        }

        return context;
    }
}

File: app/src/main/java/com/drdisagree/iconify/helpers/LocaleHelper.kt
package com.drdisagree.iconify.helpers

import android.content.Context
import android.os.LocaleList
import com.drdisagree.iconify.R
import com.drdisagree.iconify.data.common.Preferences.APP_LANGUAGE
import com.drdisagree.iconify.data.config.RPrefs.getString
import com.drdisagree.iconify.data.config.RPrefs.putString
import java.util.Locale

object LocaleHelper {

    fun setLocale(context: Context): Context {
        var localeCode = getString(APP_LANGUAGE)

        if (localeCode.isNullOrEmpty()) {
            val locales = context.resources.configuration.locales
            val localeCodes = listOf(*context.resources.getStringArray(R.array.locale_code))

            for (i in 0 until locales.size()) {
                val languageCode = locales[i].language
                val countryCode = locales[i].country
                val languageFormat = "$languageCode-$countryCode"

                if (localeCodes.contains(languageFormat)) {
                    localeCode = languageFormat
                    break
                }
            }

            if (localeCode.isNullOrEmpty()) {
                localeCode = "en-US"
            }

            putString(APP_LANGUAGE, localeCode)
        }

        val locale = Locale.forLanguageTag(localeCode)
        val configuration = context.resources.configuration
        configuration.setLocale(locale)
        configuration.setLayoutDirection(locale)

        val localeList = LocaleList(locale)
        LocaleList.setDefault(localeList)
        configuration.setLocales(localeList)

        return context.createConfigurationContext(configuration)
    }
}